File: app-android/src/main/java/com/rmakiyama/skeleton/di/CoreModule.kt
package com.rmakiyama.skeleton.di

import com.rmakiyama.skeleton.usecase.GetItemList
import com.rmakiyama.skeleton.usecase.di.UseCaseModule
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent

@Module
@InstallIn(SingletonComponent::class)
object CoreModule {

    private val useCaseModule = UseCaseModule()

    @Provides
    fun provideGetItemListUseCase(): GetItemList = useCaseModule.provideGetItemList()
}

File: app-ios/app-ios/Di/CoreModule.swift
import Foundation
import core

final class CoreModule {
    
    static let shared = CoreModule()
    
    private let useCaseModule: UseCaseModule
    private init(
        useCaseModule: UseCaseModule = UseCaseModule.companion.get()
    ) {
        self.useCaseModule = useCaseModule
    }
    
    func provideGetItemList() -> GetItemList {
        useCaseModule.provideGetItemList()
    }
}
